core/arch/arm/kernel/thread_optee_smc.c

```diff
@@ -636,11 +636,14 @@ static struct mobj *get_rpc_alloc_res(struct optee_msg_arg *arg,
 
 	if (!mobj) {
 		thread_rpc_free(bt, cookie, mobj);
-		return NULL;
+		goto out;
 	}
 
 	assert(mobj_is_nonsec(mobj));
-
+out:
```

Comment:
Since we only do `goto out` on error and there's not any common cleanup anyway, let's ahve an `err:` label instead which doesn't need to test if `mobj` is NULL or not since it always will be so.

Reply:
OK, updated